File: sample/src/main/java/com/kfaraj/support/sample/widget/SpaceItemDecoration.java
package com.kfaraj.support.sample.widget;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.Dimension;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.State;

/**
 * Adds space around each item.
 */
public class SpaceItemDecoration extends RecyclerView.ItemDecoration {

    private final int mSpace;

    /**
     * Constructor.
     *
     * @param space the amount of space around each item, in pixels.
     */
    public SpaceItemDecoration(@Dimension int space) {
        mSpace = space;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view,
            @NonNull RecyclerView parent, @NonNull State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.set(mSpace, mSpace, mSpace, mSpace);
    }

}

File: recyclerview-sample/src/main/java/com/kfaraj/support/recyclerview/sample/widget/SpaceItemDecoration.kt
package com.kfaraj.support.recyclerview.sample.widget

import android.graphics.Rect
import android.view.View
import androidx.annotation.Dimension
import androidx.recyclerview.widget.RecyclerView

/**
 * Adds [space] around each item.
 */
class SpaceItemDecoration(
    @Dimension private val space: Int
) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.set(space, space, space, space)
    }

}
